packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -349,24 +330,115 @@ const createHistoryService = ({ strapi }: { strapi: Core.Strapi }) => {
         const entryWithRelations = await Object.entries(entry.schema).reduce(
           async (currentDataWithRelations, [attributeKey, attributeSchema]) => {
+            const attributeValue = entry.data[attributeKey];
+            const attributeValues = Array.isArray(attributeValue)
+              ? attributeValue
+              : [attributeValue];
+
+            if (attributeSchema.type === 'media') {
+              return {
+                ...(await currentDataWithRelations),
+                [attributeKey]: await buildMediaResponse(attributeValues),
+              };
+            }
+
             // TODO: handle relations that are inside components
-            if (['relation', 'media'].includes(attributeSchema.type)) {
-              const attributeValue = entry.data[attributeKey];
-              const relationResponse = await buildRelationReponse(
-                (Array.isArray(attributeValue)
-                  ? attributeValue
-                  : [attributeValue]) as EntryToPopulate[],
-                attributeSchema
-              );
+            if (
+              attributeSchema.type === 'relation' &&
+              attributeSchema.relation !== 'morphToOne' &&
+              attributeSchema.relation !== 'morphToMany'
+            ) {
+              /**
+               * Don't build the relations response object for relations to admin users,
+               * because pickAllowedAdminUserFields sanitizati
+               */
+              if (attributeSchema.target === 'admin::user') {
+                const sanitizedAdminUser = await Promise.all(
+                  attributeValues.map(async (userToPopulate) => {
+                    if (userToPopulate == null) {
+                      return null;
+                    }
+
+                    const user = await strapi
+                      .query('admin::user')
+                      .findOne({ where: { id: userToPopulate.id } });
+
+                    const permissionChecker = getContentManagerService('permission-checker').create(
+                      {
+                        userAbility: params.state.userAbility,
+                        model: 'admin::user',
+                      }
+                    );
+                    const sanitizedUser = await permissionChecker.sanitizeOutput(user);
+
+                    return sanitizedUser;
+                  })
+                );
+
+                return {
+                  ...(await currentDataWithRelations),
+                  /**
+                   * TODO:
+                   *
+                   * Ideally we would return the same "{results: [], meta: {}}" shape, however,
+                   * when sanitizing the data as a whole in the controller before sending to the client,
+                   * the data for admin relation user is completely sanitized if we return an object here as opposed to an array.
+                   */
+                  [attributeKey]: sanitizedAdminUser,
+                };
```

Comment:
ℹ️ For now I think this workaround is acceptable to fix the issue and get the tests passing. 

I'd like to investigate this more in another PR like: https://github.com/strapi/strapi/pull/20179 or another refactor PR where I'd also like to move `buildMediaResponse` and `buildRelationResponse` outside of this function.

Reply:
while having to do this is a bummer, I doubt there would be a much cleaner way to do it